src/main/java/org/elasticsearch/index/shard/IndexShard.java

```diff
@@ -784,17 +786,14 @@ public RecoveryState recoveryState() {
         return this.recoveryState;
     }
 
-    public void performRecoveryFinalization(boolean withFlush) throws ElasticsearchException {
+    public void finalizeRecovery(boolean withFlush) throws ElasticsearchException {
```

Comment:
instead of passing a boolean can we maybe do the flush outside of this? I can see it being used only once if it doesn't really make a difference if it happens before the setStage call?